lib/src/line_splitting/line_splitter.dart

```diff
@@ -119,11 +119,8 @@ class LineSplitter {
   /// page width.
   LineSplitter(this.writer, this.chunks, this.blockIndentation)
       : // Collect the set of rules that we need to select values for.
-        rules = chunks
-            .map((chunk) => chunk.rule)
-            .whereType<Rule>()
-            .toSet()
-            .toList(growable: false) {
+        rules =
+            chunks.map((chunk) => chunk.rule).toSet().toList(growable: false) {
```

Comment:
```suggestion
            {for (var chunk in chunks) chunk.rule}.toList(growable: false) {
```

Reply:
It's only three characters shorter and I think the current code fits a little nicer into a method chain inside a constructor initializer, so I'll leave it as-is.